frontends/web/src/routes/device/bitbox02/createbackup.tsx

```diff
@@ -15,83 +15,70 @@
  * limitations under the License.
  */
 
-import { Component } from 'react';
-import { createBackup } from '../../../api/bitbox02';
-import { translate, TranslateProps } from '../../../decorators/translate';
+import { useState } from 'react';
+import { createBackup as createBackupAPI } from '../../../api/bitbox02';
 import { apiPost } from '../../../utils/request';
 import { alertUser } from '../../../components/alert/Alert';
 import { confirmation } from '../../../components/confirm/Confirm';
 import { Button } from '../../../components/forms';
 import { WaitDialog } from '../../../components/wait-dialog/wait-dialog';
+import { useTranslation } from 'react-i18next';
 
-interface CreateProps {
+type TProps = {
     deviceID: string;
 }
 
-type Props = CreateProps & TranslateProps;
+export const Create = ({ deviceID }: TProps) => {
+  const [creatingBackup, setCreatingBackup] = useState(false);
+  const [disabled, setDisabled] = useState(false);
+  const { t } = useTranslation();
 
-interface State {
-    creatingBackup: boolean;
-    disabled: boolean;
-}
-
-class Create extends Component<Props, State> {
-  public readonly state: State = {
-    creatingBackup: false,
-    disabled: false,
-  };
-
-  private createBackup = () => {
-    this.setState({ creatingBackup: true });
-    createBackup(this.props.deviceID)
-      .then(() => this.setState({ creatingBackup: false, disabled: false }))
+  const createBackup = () => {
+    setCreatingBackup(true);
+    createBackupAPI(deviceID)
+      .then(() => {
+        setCreatingBackup(false);
+        setDisabled(false);
+      })
       .catch(() => {
-        alertUser(this.props.t('backup.create.fail'));
+        setCreatingBackup(false);
+        setDisabled(false);
```

Comment:
this is different than before. Makes sense to me but not sure how to test, @Beerosagos can you test this?

Reply:
Yeah, this was actually a bug before. To test this, just click on "create backup"and cancel from the bb02 device. This should show a dismissable modal with "'backup.create.fail'" message. In master, there's an undismissable modal on top of it.